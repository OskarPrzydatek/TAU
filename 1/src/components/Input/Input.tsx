import React from 'react';

type InputProps = {
  label: string;
  placeholder: string;
  type?: 'text' | 'number' | 'password';
  inputTestId?: string;
  labelTestId?: string;
};

export const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ label, placeholder, type, inputTestId, labelTestId }, ref) => {
    return (
      <label>
        <span data-testid={labelTestId}>{label}</span>
        <input
          data-testid={inputTestId}
          ref={ref}
          type={type ?? 'text'}
          placeholder={placeholder}
        />
      </label>
    );
  }
);
