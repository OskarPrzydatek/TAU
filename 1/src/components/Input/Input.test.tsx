import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from './Input';

describe('Input', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<Input label="Email" placeholder="Put email here :D" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure label and placeholder renders correctly', () => {
    render(
      <Input
        label="Email"
        placeholder="Put email here :D"
        inputTestId="input"
        labelTestId="label"
      />
    );
    expect(screen.getByTestId('label')).toHaveTextContent('Email');
    expect(screen.getByTestId('input')).toHaveAttribute(
      'placeholder',
      'Put email here :D'
    );
  });

  test('ensure text is the default type of input', () => {
    render(
      <Input
        label="Email"
        placeholder="Put email here :D"
        inputTestId="input"
      />
    );
    expect(screen.getByTestId('input')).toHaveAttribute('type', 'text');
  });

  test('ensure input is writable', () => {
    render(
      <Input
        label="Email"
        placeholder="Put email here :D"
        inputTestId="input"
      />
    );
    const input: HTMLInputElement = screen.getByTestId('input');
    expect(input.value).toBe('');
    fireEvent.change(input, { target: { value: 'name@mail.com' } });
    expect(input.value).toBe('name@mail.com');
  });

  test('ensure handlig value via ref works correctly', () => {
    const inputRef = React.createRef<HTMLInputElement>();
    render(
      <Input
        ref={inputRef}
        label="Email"
        placeholder="Put email here :D"
        inputTestId="input"
      />
    );
    expect(inputRef.current?.value).toBe('');
    fireEvent.change(screen.getByTestId('input'), {
      target: { value: 'name@mail.com' },
    });
    expect(inputRef.current?.value).toBe('name@mail.com');
  });
});

export {};
