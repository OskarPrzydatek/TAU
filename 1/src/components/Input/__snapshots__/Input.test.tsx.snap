// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Snapshot 1`] = `
<label>
  <span>
    Email
  </span>
  <input
    placeholder="Put email here :D"
    type="text"
  />
</label>
`;
